### DISCUSSION TITLE
Reloading code with custom server

### DISCUSSION THREAD

**Discussion by savil** [vercel/next.js#12972] [Help] [answered]
Hello there - i have a custom server I'm using with NextJS for the client-side frontend. I'm trying to have my code reload quickly during development. From #791, `nodemon` seems to be the solution for the server code. However, to use any changes in the client-side frontend in `pages/`, it seems I still need to do `next build`  --- which is achingly slow for me.

Is there a configuration that achieves the combination of `next dev` for the frontend changes and `nodemon` for the custom-server changes?

**Comment by jamesmosier** [comment, marked as answer]
You should be able to use something [like this custom server example](https://github.com/zeit/next.js/tree/canary/examples/custom-server) and instead of running `node server.js` like [in the example package.json](https://github.com/zeit/next.js/blob/01211ba21de39a8ac9307cb23cbfa7b0d5fdbc4d/examples/custom-server/package.json#L5) you can instead just do `"dev": "nodemon server.js"` and now you'll have code reloading for your server as well as for your frontend code.
